### DISCUSSION TITLE
Public key in self hosting keygen

### DISCUSSION THREAD

**Discussion by abdur-tech** [keygen-sh/community#124] [Issue]
Hi team,

I am integrating self hosting Keygen in my personal application, I need to implement offline licensing. I have created license using scheme attribute as ed25519_SIGN. Now I need validate the licence key in offline. How to get keygen_public_key from self hosting Keygen server.

**Comment by ezekg** [comment]
You can get it via [the console](https://keygen.sh/docs/self-hosting/#opening-a-console):

```rb
Account.sole.ed25519_public_key
```

**Reply by abdur-tech** [reply]
Appreciate the assistance! Could you guide me on acquiring this command, or is there specific documentation I should consult?

**Reply by ezekg** [reply]
I linked to the docs: https://keygen.sh/docs/self-hosting/#opening-a-console

Instead of running a container with the command `keygen/api web`, run it with `keygen/api console`.

Or if you're on the server, run `rails console` in the app directory.
